### ISSUE TITLE
in-project stubs cause mypy to bail in 0.971

### ISSUE THREAD

**Issue by vfazio** [python/mypy#13214]
**Bug Report**

We have a project that includes type stubs for a python package that does not provide it's own typing. We followed guidance per https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-library-stubs-or-py-typed-marker number 3

```
(venv) vfazio@vfazio2 ~/development/xtf $ grep mypy_path mypy.ini
mypy_path=$MYPY_CONFIG_FILE_DIR/stubs
```

In mypy 0.961, type checks pass

```
(venv) vfazio@vfazio2 ~/development/xtf $ mypy -V; mypy xes/
mypy 0.961 (compiled: no)
Success: no issues found in 86 source files
```
In mypy 0.971, these now fail because the current path is being added to the path search 

```
(venv) vfazio@vfazio2 ~/development/xtf :( $ mypy -V; mypy xes/
mypy 0.971 (compiled: no)
['/mnt/development/mypy/venv/lib/python3.10/site-packages', '/mnt/development/xtf']
/mnt/development/xtf is in the MYPYPATH. Please remove it.
See https://mypy.readthedocs.io/en/stable/running_mypy.html#how-mypy-handles-imports for more info

```

If i modify `mypy/pyinfo.py::getsearchdirs` to exclude the current working directory (effectively reverting 2132036a019ed26d77389c0a9af9b928911a5846) the type checks "work" except I now get type failures for the stubs

```
(venv) vfazio@vfazio2 ~/development/xtf $ mypy -V; mypy xes/
mypy 0.971 (compiled: no)
stubs/redacted/path/file.pyi:49:9: error: Function is missing a return type annotation  [no-untyped-def]
stubs/redacted/path/file.pyi:49:9: note: Use "-> None" if function does not return a value
<snip>
Found 58 errors in 2 files (checked 86 source files)

```

**Expected Behavior**

Failures wouldn't occur or documentation would be updated to reflect how to accommodate this use case.

**Actual Behavior**

mypy now fails to type check our code because we include stubs as part of our source tree

**Your Environment**

- Mypy version used: 0.971
- Mypy command-line flags:
- Mypy configuration options from `mypy.ini` (and other config files):
- Python version used: 3.10.4
- Operating system and version: Ubuntu 20.04

**Comment by hauntsaninja** [comment]
And thanks for the report and the extra investigation, makes it much easier to figure out what's going on!

**Comment by vfazio** [comment]
Ok, I think I kinda-sorta understand what's going on.

As a test, I reverted my hack to exclude `cwd` and dropped the `sys.exit(1)` from modulefinder after the MYPYPATH warning and I now see that the .pyi files no longer trigger warnings. 

For now, we'll stick to 0.961 until there's a patch to address this behavior.

It may make sense to add a unittest for this scenario since it's a documented configuration, that way any future regressions could be caught?

**Comment by uSpike** [comment]
I think I've found the specific environment parts that are causing our error.

* We use poetry
* We have a namespace package and use the `packages` feature: https://python-poetry.org/docs/pyproject/#packages
* `mypy.ini` contains `mypy_path=$MYPY_CONFIG_FILE_DIR/stubs`

So our pyproject.toml looks like:
```
[tool.poetry]
name = "test_project"
...
packages = [
    { include = "namespace_pkg" }
]
```

Upon `poetry install`, a `test_project.pth` file is created in `<virtualenv path>/lib/python3.10/site-packages/`:
```
 $ ls <redacted>/lib/python3.10/site-packages/ -la                                                                                                                        
...                                                            
-rwxr-xr-x  1 jspeicher domain users 45790632 Jul 22 11:19 f4771e8a540c98c8e0df__mypyc.cpython-310-x86_64-linux-gnu.so
drwxr-xr-x  9 jspeicher domain users    12288 Jul 22 11:19 mypy
drwxr-xr-x  2 jspeicher domain users     4096 Jul 22 11:19 mypy-0.971.dist-info
drwxr-xr-x  2 jspeicher domain users     4096 Jul 22 11:19 mypy_extensions-0.4.3.dist-info
-rw-r--r--  1 jspeicher domain users     5078 Jul 22 11:19 mypy_extensions.py
drwxr-xr-x 14 jspeicher domain users     4096 Jul 22 11:19 mypyc
...
drwxr-xr-x  2 jspeicher domain users     4096 Jul 22 11:20 test_project-0.1.0.dist-info
-rw-r--r--  1 jspeicher domain users       30 Jul 22 11:20 test_project.pth
...
```

With this `test_project.pth` file, mypy errors as @vfazio commented.
```
$ mypy namespace_pkg
<project dir> is in the MYPYPATH. Please remove it.
```
If I comment-out the `packages` portion of `pyproject.toml` then the `test_project.pth` file is not created, and mypy works as expected.

---

I also tried this with a simple setuptools-based setup:

setup.py:
```python
from setuptools import setup

setup(
   name='test_project',
   version='2.0',
   packages=['namespace_pkg'],
)
```

Upon `pip install -e .`, a `venv/lib/python3.10/site-packages/test-project.egg-link` file is created, which causes the same mypy error:
```
$ mypy namespace_pkg
<project dir> is in the MYPYPATH. Please remove it.
```

**Comment by uSpike** [comment]
To sum up my previous comment, I think that mypy is currently incompatible with an editable-installed package (`pip install -e .`) in combination with mypy config `mypy_path=$MYPY_CONFIG_FILE_DIR/<anything>`

**Comment by sobolevn** [comment]
Today I am solving the ~same problem: https://github.com/typeddjango/django-stubs/pull/1108

Mypy 0.960 works just fine, but 0.971 just does not see my stubs 😢
